### ISSUE TITLE
Help: Help for triggering the record function in Queue

### ISSUE THREAD

**Issue by hasnat-we** [cyrildewit/eloquent-viewable#255]
Thanks a lot for this plugin. I was trying to fire the record method in a Queue job like below in Controller.
`EloquentViewableVisitLogger::dispatchAfterResponse($post);`

It increase the view record nicely but the cooldown is not working as expected since session is missing from Queue environment. So the views count is increasing for each visit and page load.

Can you please give me a suggestion, is there any way to make cooldown work in Queue environment? Thanks in advance.

**Comment by cyrildewit** [comment]
This is a known limitation of the design. The session check should be done before adding the job to the queue, but since this logic is tightly coupled to the implementation, there is currently no way to do this beforehand. If you've any ideas to improve the design, please let me know.

For your information, I'm currently not actively adding new features or making improvements. I do accept contributions and I'm open to discussing new ideas. In the future when I have more time, I will probably do an iteration again to fix some of these problems.

**Comment by cyrildewit** [comment]
Hi @hasnat-we, thanks for your patience. Native support for queueing view recording has been added in v8. You can now defer the database write to a queue while all filtering (crawler detection, Do Not Track, ignored IPs, and **cooldowns**) still runs during the request:

```php
views($post)->queue()->record();
```

Because the cooldown check happens in the request and not in the worker, the session is still available, so duplicate views are correctly skipped. Only the insert is deferred to the queue. You can also enable queueing globally via the `queue` option in the config file. See the "Queueing view recording" section of the README for details.
